Assets/MixedRealityToolkit.Extensions/SceneTransitionService/Scripts/CameraFaderQuad.cs

```diff
@@ -0,0 +1,204 @@
+        }
+
+        public CameraFaderState State { get; private set; }
+
+        private Dictionary<Camera, Quad> quads = new Dictionary<Camera, Quad>();
+        private Color fadeOutColor;
+        private Color fadeInColor;
+        private Color currentColor;
+        private Material quadMaterial;
+
+        public async Task FadeOutAsync(float fadeOutTime, Color color, IEnumerable<Camera> targets)
+        {
+            switch (State)
+            {
+                case CameraFaderState.Clear:
+                    break;
+
+                default:
+                    Debug.LogWarning("Can't fade out in state " + State + " - not proceeding.");
+                    return;
+            }
+
+            State = CameraFaderState.FadingOut;
+
+            fadeOutColor = color;
+            fadeInColor = fadeOutColor;
+            fadeInColor.a = 0;
+
+            if (fadeOutColor.a < 1)
+            {
+                Debug.LogWarning("Target color is not fully opaque.");
+            }
+
+            await Task.Yield();
+
+            // Create our material
+            if (quadMaterial == null)
+            {
+                try
+                {
+                    quadMaterial = new Material(Shader.Find(QuadMaterialShaderName));
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError("Error when trying to create quad material in CameraFaderQuad");
+                    Debug.LogException(e);
+                    return;
+                }
+            }
+
+            quadMaterial.SetColor(QuadMaterialColorName, currentColor);
+
+            // Create our quads
+            foreach (Camera camera in targets)
+            {
+                // Can't target the same camera twice
+                if (quads.ContainsKey(camera))
+                    continue;
+
+                Quad quad = new Quad();
+                quad.PropertyBlock = new MaterialPropertyBlock();
+                quad.Renderer = GameObject.CreatePrimitive(PrimitiveType.Quad).GetComponent<Renderer>();
+                quad.Renderer.sharedMaterial = quadMaterial;
+                // Parent the quad under the camera
+                quad.Renderer.transform.parent = camera.transform;
+                quad.Renderer.transform.localScale = camera.orthographic ? Vector3.one * camera.orthographicSize : Vector3.one * camera.fieldOfView;
+                quad.Renderer.transform.localPosition = Vector3.forward * camera.nearClipPlane * 1.01f;
+                quad.Renderer.transform.localRotation = Quaternion.identity;
+
+                // Set the quad's layer to something the camera sees
+                for (int layer = 0; layer < 32; layer++)
+                {
+                    if (camera.cullingMask == (camera.cullingMask | (1 << layer)))
+                    {
+                        quad.Renderer.gameObject.layer = layer;
+                        break;
+                    }
+                }
```

Comment:
Is it just because of how transient this quad is going to be that it doens't matter which layer gets picked here? It seems arbitrary here (i.e. we could assign it to a layer that is reserved for app specific stuff)

Reply:
Correct. As long as it's seen by the camera it'll work.